resources/views/engagements/show.blade.php

```diff
@@ -59,9 +59,16 @@
             </dl>
 
             @can('join', $engagement)
-                <a class="cta" href="{{ localized_route('engagements.sign-up', $engagement) }}">
-                    @svg('heroicon-o-clipboard-check') {{ __('Sign up') }}
-                </a>
+                <div class="flex flex-col">
+                    <a class="{{ $engagement->confirmedParticipants->count() >= $engagement->ideal_participants ? 'cta disabled' : 'cta' }} mx-auto"
+                        href="{{ localized_route('engagements.sign-up', $engagement) }}"
+                        tabindex="{{ $engagement->confirmedParticipants->count() >= $engagement->ideal_participants ? '-1' : '0' }}">
+                        @svg('heroicon-o-clipboard-check') {{ __('Sign up') }}
+                    </a>
+                    @if ($engagement->confirmedParticipants->count() >= $engagement->ideal_participants)
+                        <p>{{ __('All participant spots have been filled.') }}</p>
+                    @endif
+                </div>
```

Comment:
Another issue is that the disabled state will never show up. The `@can` directive used on line 61 will prevent the sign up button from showing if the capacity has been reached. 

If we still need to show a disabled "sign up" link, we may need to do a check for all the following to decide if it is displayed:

- cannot "join"
- cannot "participate" (see line 74 below)
- engagement confirmed count is greater than or equal to the ideal number of participants.

Reply:
I realized that join policy already includes check for the participants count; I updated the policy to not to check the participants counts, because we want to have all the other rules specified for the join. Also, I don't think we need to consider case of cannot participate.